### PR TITLE
feat: Repository search extn

### PR DESCRIPTION
**Pull request by shamimbinnur**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR [adds/removes/fixes/replaces] the [feature/bug/etc]. 

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: src/components/PrimaryNav.tsx [hunk 1]
```diff
@@ -1,23 +1,137 @@
...
+import useSupabaseAuth from "../hooks/useSupabaseAuth";
+import { version } from "../../package.json";
+import { fetchWithSearch } from "../lib/supabase";
 
-const PrimaryNav = (): JSX.Element => {
+interface PostWrapProps{
+  setTextToSearch: any
+}
+
+// TODO: move to hooks/debounce.ts
```
Comment (0-vortex) [inline src/components/PrimaryNav.tsx:12]: use https://usehooks.com/useDebounce/ please or the radix example one :D

Comment (shamimbinnur) [reply src/components/PrimaryNav.tsx:12]: Thanks for reviewing, This hook will be called only when the user clicks on the repository item from the input dropdown items. So, it is not going to be called frequently. Do we still need the denouncing here?

**Comment by bdougie** [comment]
This is great, thanks for taking this on from #91.

When searching, when the user clicks out of the search box or loses focus, the box stays present. I would expect it to disappear or be remove when click out of the search box.
 
Screen Shot 2022-05-25 at 6 05 22 AM

**Comment by shamimbinnur** [comment]
> Just noticed this. Could you add some padding on the right @shamimbinnur. The words go right up on the edge. Screen Shot 2022-05-25 at 9 06 03 AM

@bdougie 
Sorry for the inconvenience. Added padding and hover effect. Can you please check now?

**Comment by bdougie** [comment]
I don't think this was from your changes, but looks like unused variables.

**Comment by bdougie** [comment]
Ok after some chats with the Open Sauced community on stream here are some updates. 
1. Search should not filter the results on the page when clicked. Instead the user should navigate to "https://app.opensauced.pizza/repos/{fullName}, when full name the vuejs/vue or similar.
2. Need to address the errors with a `onKeydown` event.

```
role="button"
                                tabIndex={0}
                                aria-pressed="false"
                                className=" text-grey text-xs sm:text-lg px-[15px] py-[10px] font-medium overflow-hidden cursor-pointer hover:bg-gray-200 "
                                onClick={()=> {
                                  clickHandler(result.full_name)
                                }}
                                onKeyDown={async (e)=> {
                                  if (e.key === 'Enter') {
                                    await clickHandler(result.full_name)
                                  }
                                }}
                                ```

**Comment by bdougie** [comment]
Merged my branch her. There are some small errors in accessibility still and tabbing doesn't quite work, but we can merge this into the beta branch instead of main and tackle those other things as separate issues.

**Pull request closed without merging**